ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_protection_template.py

```diff
@@ -0,0 +1,389 @@
+    type: str
+    description:
+    - Application server hostname.
+  app_service_name:
+    required: False
+    type: str
+    description:
+    - If the application is running within a windows cluster environment then this is the instance name of the service running within the cluster environment.
+  app_sync:
+    choices:
+        - none
+        - vss
+        - vmware
+        - generic
+    required: False
+    type: str
+    default: none
+    description:
+    - Application synchronization.
+  change_name:
+    required: False
+    type: str
+    description:
+    - Change the name of existing protection template.
+  description:
+    required: False
+    type: str
+    description:
+    - Text description of protection template.
+  name:
+    required: True
+    type: str
+    default: None
+    description:
+    - Name of protection template.
+  state:
+    required: True
+    choices:
+        - present
+        - absent
+        - create
+    type: str
+    description:
+    - Choice for protection template operations.
+  vcenter_hostname:
+    required: False
+    type: str
+    description:
+    - VMware vCenter hostname.
+  vcenter_password:
+    required: False
+    type: str
+    description:
+    - Application VMware vCenter password. A password with few constraints.
+  vcenter_username:
+    required: False
+    type: str
+    description:
+    - Application VMware vCenter username. String of up to 80 alphanumeric characters, beginning with a letter.
+      It can include ampersand (@), backslash (\), dash (-), period (.), and underscore (_).
+extends_documentation_fragment: hpe_nimble
+short_description: Manage HPE Nimble Storage protection template.
+version_added: 2.9
+'''
+
+EXAMPLES = r'''
+
+# if state is create , then create a protection template if not present. Fails if already present.
+# if state is "present", then create a protection template if not present. Succeed if it already exists.
+- name: Create protection template if not present
+  hpe_nimble_protection_template:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    name: "{{ name }}"
+    description: "{{ description | default(None)}}"
+    state: "{{ state | default('present') }}"
+
+- name: Delete protection template
+  hpe_nimble_protection_template:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    name: "{{ name }}"
+    state: absent
+
+'''
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from nimbleclient.v1 import client
+except ImportError:
+    client = None
+import ansible_collections.hpe.nimble.plugins.module_utils.hpe_nimble as utils
+
+def create_prot_template(
+        client_obj,
+        prot_template_name,
+        **kwargs):
+
+    if utils.is_null_or_empty(prot_template_name):
+        return (False, False, "Create protection template failed as volume collection is not present.", {})
```

Comment:
Error message references the volume collection.  This should be the name.

Reply:
fixed